src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,347 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Actions {
+
+inline constexpr size_t kActionNameMaxSize       = 128u;
+inline constexpr size_t kEndpointListNameMaxSize = 128u;
+
+class Delegate;
+
+struct GenericActionStruct : public Structs::ActionStruct::Type
```

Comment:
This is not "Generic" in any sense (unlike for Mode Base, where things are generic over derived clusters).

Shouldn't this just be ActionStructWithLabelStorage or something?